llvm/lib/Target/Sparc/SparcCallingConv.td

```diff
@@ -34,6 +34,7 @@ def RetCC_Sparc32 : CallingConv<[
   CCIfType<[i32], CCAssignToReg<[I0, I1, I2, I3, I4, I5]>>,
   CCIfType<[f32], CCAssignToReg<[F0, F1, F2, F3]>>,
   CCIfType<[f64], CCAssignToReg<[D0, D1]>>,
+  CCIfType<[f128], CCCustom<"CC_Sparc_Assign_Ret_F128">>,
```

Comment:
Will this work (without the custom assigner)?
```suggestion
  CCIfType<[f128], CCIfInReg<CCAssignToReg<[Q0, Q1]>>>,
```

Reply:
Ah TIL there's this, thanks.
Though I have a question, is it possible to only assign registers to it if the f128 is part of a `{f128,f128}`?
If possible, I want to tighten the conditionals, but if not then just checking for inreg should be fine.

Reply:
CCIfConsecutiveRegs, probably